library/lib/withPolyfill.js

```diff
@@ -0,0 +1,53 @@
+const { getPolyfillString } = require('polyfill-service')
+const pathname = '/polyfill.js'
+const pathnameMinified = '/polyfill.min.js'
+
+export default function withPolyfill(Wrapped) {
+  WithPolyfill.routes = {
+    match: async (location, request) => {
+      const uaString = request.get('User-Agent')
+
+      if ([pathname, pathnameMinified].includes(location.pathname)) {
+        const minify = location.pathname === pathnameMinified
+        return polyfillResponse({ uaString, query: request.query, minify })
+      }
+
+      if (Wrapped.routes) {
+        const { status, headers, data: originalData } = await Wrapped.routes.match(location, request)
+        return { status, headers, data: { uaString, originalData }}
+      }
+
+      return { status: 200, data: null }
+    }
+  }
+
+  function WithPolyfill({ location, data: { polyfill, uaString, originalData } }) {
+    return polyfill || <Wrapped location={location} data={originalData} polyfill={getPolyfill} />
+
+    function getPolyfill(features = [], { minify = true } = {}) {
+      const path = minify ? pathnameMinified : pathname
+      return (<script defer src={`${path}?features=${features.join(',')}&cache=${encodeURIComponent(uaString)}`} />)
+    }
+  }
+
+  return WithPolyfill
+}
+
+
+async function polyfillResponse({uaString, minify, query = {} }) {
+  const features = (query.features || '').split(',').filter(Boolean).reduce((res, x) => ({ ...res, [x]: { flags: ['gated'] }}), {})
+  const cache = !!query.cache
+
+  const polyfill = await getPolyfillString({ uaString, minify, features })
+  const headers = {
+    'Content-Type': 'application/javascript;charset=utf-8',
+    'Content-Length': polyfill.length,
```

Comment:
This is incorrect in case of a non ascii character (if you have a character that takes more than one byte) in the polyfill. I thought you tested with the `express.static` plugin and found that it did not specify `Content-Length` for `.js` files.

Reply:
mark